cmake/onnxruntime_hosting.cmake

```diff
@@ -24,6 +24,12 @@ target_include_directories(${PROJECT_NAME} PRIVATE
 
 target_link_libraries(${PROJECT_NAME} PRIVATE
         ${Boost_LIBRARIES}
+        ${PROVIDERS_MKLDNN}
```

Comment:
Can we delete any of these?

Reply:
without the use_xxx turned on, they will be empty. But we may end up adding them any way to build onnx runtime in different way. So I will keep them as-is and clean them up when those 3rd party libraries are not used by onnx runtime in any form.